src/app/components/week-selector/week-selector.component.ts

```diff
@@ -0,0 +1,69 @@
+import { async } from '@angular/core/testing';
+import { Component, OnInit, Input, Output, EventEmitter, AfterContentChecked } from '@angular/core';
+import { addWeeks, subWeeks, subYears, getWeek, getYear } from 'date-fns';
+
+@Component({
+  selector: 'app-week-selector',
+  templateUrl: './week-selector.component.html',
+  styleUrls: ['./week-selector.component.scss']
+})
+export class WeekSelectorComponent implements OnInit {
+
+  @Input() currentDate: Date;
+  @Input() startDate: Date;
+  @Input() subTitle: string;
+
+  @Output() passDate = new EventEmitter<number>();
+
+  weekNumber: number;
+  year: number;
+  isCurrentWeek: boolean;
+  isStartingWeek: boolean;
+  disabledLeft: boolean = false;
+  disabledRight: boolean;
+  date: Date
+
+  constructor() {}
+
+  ngOnInit() {
+    this.isCurrentWeek = true;
+    this.disabledRight = true;
+    this.date = this.currentDate
+    this.weekNumber = getWeek(this.date);
+    this.passDate.emit(this.weekNumber)
+    this.year = getYear(this.date)
+  }
+
+  onPreviousWeek() {
+    this.date = subWeeks(this.date, 1);
+    this.weekNumber = getWeek(this.date)
+    
+    this.setCurrentWeekBoolean();
+    
+    
+    this.disabledRight = false;
+    this.disabledLeft = this.isDateBeforeStartDate();
+    
+    this.passDate.emit(this.weekNumber);
+  }
+
+  onNextWeek() {
+    this.date = addWeeks(this.date, 1);
+    this.weekNumber = getWeek(this.date)
+    
+    this.passDate.emit(this.weekNumber);
+    
+    this.disabledLeft = false;
+    this.isStartingWeek = false
+    
+    this.disabledRight = this.setCurrentWeekBoolean();
+  }
+
+  private setCurrentWeekBoolean() {
+    return getYear(this.date) === getYear(this.currentDate) && getWeek(this.date) === getWeek(this.currentDate);
+  }
+
+  private isDateBeforeStartDate() {
```

Comment:
Duidelijke naam! Als overdenking: ik zou bij deze naam een variable verwachten en geen method. Zelf maak ik altijd onderscheid daartussen op deze manier:

`
const isDateBeforeStartDate = this.checkIsDateBeforeStartDate();
`

Maar puur ter overdenking, want hij is prima duidelijk zo 👍

Reply:
Nu je het zegt ik zou daar egenlijk ook een variabelen verwachten.. Ga ik even over na denken.